From 1b090b2e8c4b2865101119a2938ee4c402523aea Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Fri, 31 Mar 2017 14:01:22 +0800
Subject: [PATCH 53/62] live install supports kickstart

While live install, WRLinux does not support harddrive,
use cdrom to replace

Upstream-Status: Inappropriate [wrlinux specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 anaconda              | 2 +-
 pyanaconda/install.py | 4 ++++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/anaconda b/anaconda
index 94e1268..0fe39a3 100755
--- a/anaconda
+++ b/anaconda
@@ -1047,7 +1047,7 @@ if __name__ == "__main__":
     # execute) that now.  Otherwise, load in defaults from kickstart files
     # shipped with the installation media.
     ksdata = None
-    if opts.ksfile and not opts.liveinst:
+    if opts.ksfile:
         if not os.path.exists(opts.ksfile):
             stdoutLog.error("Kickstart file %s is missing.", opts.ksfile)
             iutil.ipmi_report(constants.IPMI_ABORTED)
diff --git a/pyanaconda/install.py b/pyanaconda/install.py
index c0fc98e..7b72fe0 100644
--- a/pyanaconda/install.py
+++ b/pyanaconda/install.py
@@ -56,6 +56,10 @@ def _writeKS(ksdata):
     # Do not support network configuration
     ksdata.network.network = []
 
+    # Correct method of live install, use cdrom to replace
+    if ksdata.method.method == "harddrive":
+        ksdata.method.method = "cdrom"
+
     with open(path, "w") as f:
         f.write(str(ksdata))
 
-- 
2.7.4

